src/main.rs

```diff
@@ -115,5 +130,25 @@ async fn main() {
                 }
             }
         }
+
+        Opt::RepoParticipants {
+            org,
+            repo,
+            verbose: _,
+        } => {
+            let (tx, mut rx) = mpsc::channel::<Vec<String>>(400);
+            let repo_participants = RepoParticipants::new(org, repo, 30);
+            let column_names = repo_participants.column_names();
+
+            tokio::spawn(async move {
+                if let Err(e) = repo_participants.producer_task(tx).await {
+                    println!("Encountered an error while collecting data: {}", e);
+                };
+            });
+
+            if let Err(e) = Print::consume(Print, &mut rx, column_names).await {
+                println!("Error while printing results: {}", e);
+            }
```

Comment:
seeing all this makes me wish we had a better way to not repeat. This logic is likely going to look familiar for every command.

certainly at least moving `let (tx, mut rx)` declaration up above match statement since each command will use it?

Maybe anything beyond that would result in overcomplicating things... i don't know.

Reply:
Yes, that is exactly what was motivating https://github.com/optopodi/optopodi/issues/26 -- I think we should completely separate out the "producer" logic from the "consumer", right now they are entangled.

Reply:
understood. It’s funny… when we were making those traits it felt like a good step towards separation, but I very much see what you mean on the entanglement front. 

maybe #26 will be my next issue to tackle. (though I do wish we had tests…)